plugin/dnstap/io.go

```diff
@@ -104,21 +114,36 @@ func (d *dio) close() { close(d.quit) }
 
 func (d *dio) write(payload *tap.Dnstap) error {
 	if d.enc == nil {
-		atomic.AddUint32(&d.dropped, 1)
-		return nil
+		return errors.New("dnstap not connected to output socket")
```

Comment:
[performance] it's better to create the error object once, assign to a global var and return here, rather than re-creating it every time when hitting the error condition.